### PR TITLE
Added Resources class for resource loading from URLs

### PR DESCRIPTION
**Pull request by dorzey**
Addressed the feedback on https://github.com/joel-costigliola/assertj-core/pull/280 and moved the work done by @Turbo87

### CODE REVIEW

**Comment by Turbo87** [comment]
awesome, thanks! 

Just one little remark: the Resources class actually does no resource loading at all and only operates on URLs/Streams, so it might be better to call it `URLs` instead. I would expect the Resources class instead to accept a string and do the ClassLoader calls for me. Obviously that can't be exported in the Assertions class because it would conflict with the file loading methods, but I think it would still be useful.

**Comment by joel-costigliola** [comment]
Good point indeed, let's go for `URLs`.

**Comment by dorzey** [comment]
Renamed. Is there anything else that needs addressing?

**Comment by Turbo87** [comment]
I think this is good to merge, but as I mentioned in the last comment it would be cool to have something like the following too:

```java
public class Resources {
    public static String contentOf(String resource) {
        return URLs.contentOf(ClassLoader.getSystemResource(resource));
    }

    // ...
}
```

```java
String resource = Resources.contentOf("some-resource.json");
```

**Comment by joel-costigliola** [comment]
@Turbo87 well why not but let's remember that assertj is an assertion library not a IO utils lib.

**Comment by Turbo87** [comment]
My plan is to use it roughly like this:

```java
assertThat(result).isEqualTo(Resources.contentOf("test1.json"))
```

**Comment by joel-costigliola** [comment]
In that case, we could introduce a new String assertions:
```java
assertThat(result).isEqualToContentOfResource("test1.json");
assertThat(result).isEqualToContentOfURL("htpp://foo.com/tests/test1.json");
assertThat(result).isEqualToContentOfFile(test1.json);
```

**Comment by dorzey** [comment]
Can this be merged? And we raise a separate issue for the extensions suggested.

**Comment by Turbo87** [comment]
:+1: from my side

File: src/main/java/org/assertj/core/api/Assertions.java [hunk 1]
```diff
@@ -1211,6 +1214,89 @@ public static String contentOf(File file) {
   }
 
   // --------------------------------------------------------------------------------------------------
+  // URL/Resource methods : not assertions but here to have a single entry point to all AssertJ features.
+  // --------------------------------------------------------------------------------------------------
+
+  /**
+   * Loads the text content of a URL, so that it can be passed to {@link #assertThat(String)}. <p> Note that this will
+   * load the entire file in memory. </p>
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/Assertions.java]: Javadoc should not mention file.

File: src/main/java/org/assertj/core/api/Assertions.java [hunk 2]
```diff
@@ -1211,6 +1214,89 @@ public static String contentOf(File file) {
...
+   *
+   * @param url the URL.
+   * @param charset the character set to use.
+   * @return the content of the file.
+   * @throws NullPointerException if the given charset is {@code null}.
+   * @throws FilesException       if an I/O exception occurs.
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/Assertions.java]: let's rename `FilesException` to `RuntimeIOException`, URL are not only about file.

File: src/main/java/org/assertj/core/api/Assertions.java [hunk 3]
```diff
@@ -1211,6 +1214,89 @@ public static String contentOf(File file) {
...
+  // URL/Resource methods : not assertions but here to have a single entry point to all AssertJ features.
+  // --------------------------------------------------------------------------------------------------
+
+  /**
+   * Loads the text content of a URL, so that it can be passed to {@link #assertThat(String)}. <p> Note that this will
+   * load the entire file in memory. </p>
+   *
+   * @param url the URL.
+   * @param charset the character set to use.
+   * @return the content of the file.
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/Assertions.java]: Idem.

File: src/test/java/org/assertj/core/util/URLs_contentOf_Test.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    assertFalse(missingFile.exists());
+
+    thrown.expect(FilesException.class);
+    URLs.contentOf(missingFile.toURI().toURL(), Charset.defaultCharset());
+  }
+
+  @Test
+  public void should_load_resource_from_url_using_charset() {
+    // NB: UTF-8 must be supported by every Java implementation
+    assertEquals(expectedContent, URLs.contentOf(sampleResourceURL, Charset.forName("UTF-8")));
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/util/URLs_contentOf_Test.java]: use `assertThat`

**Comment by joel-costigliola** [comment]
Just minor comments but otherwise PR is ok.

**Comment by joel-costigliola** [comment]
Integrated, thx !

**Pull request closed without merging**